### PR TITLE
Add additional unit tests for Azure cloud provider.

### PR DESCRIPTION
**Pull request by brendandburns**
@feiskyer @andyzhangx @khenidak

### CODE REVIEW

File: pkg/cloudprovider/providers/azure/azure_backoff.go [hunk 1]
```diff
@@ -396,8 +396,8 @@ func processRetryResponse(resp autorest.Response, err error) (bool, error) {
 		// suppress the error object so that backoff process continues
 		return false, nil
 	}
-	// Fall-through: stop periodic backoff, return error object from most recent request
-	return true, err
+	// Fall-through: stop periodic backoff
+	return true, nil
```
Comment (andyzhangx) [inline pkg/cloudprovider/providers/azure/azure_backoff.go]: @brendandburns is there any explaination for this? bug fix?

File: pkg/cloudprovider/providers/azure/azure_backoff_test.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
+package azure
```
Comment (feiskyer) [inline pkg/cloudprovider/providers/azure/azure_backoff_test.go]: missing boilerplate for new files

Comment (feiskyer) [reply pkg/cloudprovider/providers/azure/azure_backoff_test.go]: Also should run `hack/update-bazel.sh` for newly added file

File: pkg/cloudprovider/providers/azure/azure_backoff.go (hunk 1 above)
Comment (feiskyer) [reply pkg/cloudprovider/providers/azure/azure_backoff.go]: @andyzhangx Yep a bug fix. Retry cases has been processed by `shouldRetryAPIRequest`, here should just stop backoff.

File: pkg/cloudprovider/providers/azure/azure_backoff_test.go (hunk 2 above)
Comment (brendandburns) [reply pkg/cloudprovider/providers/azure/azure_backoff_test.go]: Both done.

**Comment by brendandburns** [comment]
Comments addressed, rebased, please take another look....

File: pkg/cloudprovider/providers/azure/azure_backoff_test.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
```
Comment (feiskyer) [inline pkg/cloudprovider/providers/azure/azure_backoff_test.go]: Seems some blank lines missed?

**Review by feiskyer — APPROVED** [review]
/lgtm

**Pull request merged**